### PR TITLE
Remove unused version_year from script cache key

### PR DESCRIPTION
**Pull request by wjordan**
Fixes a bug introduced in #22887 where scripts fetched for a non-nil `version_year` (which included all fetches from `ScriptLevelsController#show`) were not pre-loaded during application initialization. This caused the first request (within each process) to `ScriptLevelsController#show` for each script to be slower than expected on startup.

Since #28683, `version_year` is no longer used by the script-cache fetch anyway, so all of the logic extending the cache key by this variable can be removed for simplicity.

### CODE REVIEW

File: dashboard/app/models/script.rb [hunk 1]
```diff
@@ -433,28 +433,21 @@ def self.get_without_cache(id_or_name, version_year: nil, with_associated_models
...
 
-    return get_without_cache(id_or_name, version_year: version_year, with_associated_models: false) unless should_cache?
-    cache_key_suffix = version_year ? "/#{version_year}" : ''
-    cache_key = "#{id_or_name}#{cache_key_suffix}"
-    script_cache.fetch(cache_key) do
+    return get_without_cache(id_or_name, with_associated_models: false) unless should_cache?
+    script_cache.fetch(id_or_name) do
       # Populate cache on miss.
-      script_cache[cache_key] = get_without_cache(id_or_name, version_year: version_year)
+      script_cache[cache_key] = get_without_cache(id_or_name)
```
Comment (davidsbailey) [inline dashboard/app/models/script.rb:451]: looks like cache_key needs to be defined and must be a string

File: dashboard/app/models/script.rb [hunk 2]
```diff
@@ -433,28 +433,21 @@ def self.get_without_cache(id_or_name, version_year: nil, with_associated_models
...
-  # name, or a redirect to the latest version of the script within the specified
-  # family. Also populates the script cache so that future responses will be cached.
+  # name. Also populates the script cache so that future responses will be cached.
   # For example:
   #   get_from_cache('11') --> script_cache['11'] = <Script id=11, name=...>
   #   get_from_cache('frozen') --> script_cache['frozen'] = <Script name="frozen", id=...>
-  #   get_from_cache('csp1') --> script_cache['csp1'] = <Script redirect_to="csp1-2018">
-  #   get_from_cache('csp1', version_year: '2017') --> script_cache['csp1/2017'] = <Script redirect_to="csp1-2017">
   #
   # @param id_or_name [String|Integer] script id, script name, or script family name.
```
Comment (davidsbailey) [inline dashboard/app/models/script.rb:441]: this should just be `script id or script name.`

**Review by davidsbailey — APPROVED** [review]
Thank you, @wjordan ! Looks great.

While you are in here, you could also delete the unused `get_script_family_redirect` if you want -- nothing references it, so it should be pretty low risk.

The only other thing I can think of would be to add an integration test which counts the number of database requests. However, it looks like we already have some which do that, so I'm a bit puzzled as to why those haven't been failing: https://github.com/code-dot-org/code-dot-org/blob/06d2450238029c0e94d05e1838ed657672e33a5d/dashboard/test/integration/caching_test.rb#L22-L27

**Comment by davidsbailey** [comment]
> The only other thing I can think of would be to add an integration test which counts the number of database requests. However, it looks like we already have some which do that, so I'm a bit puzzled as to why those haven't been failing

Oops, bad example, because frozen is a HOC script. but course1 is not, so I would have expected this test to have been failing: https://github.com/code-dot-org/code-dot-org/blob/06d2450238029c0e94d05e1838ed657672e33a5d/dashboard/test/integration/caching_test.rb#L80-L85

**Comment by davidsbailey** [comment]
ok... looks like those tests take a cache miss the first time, and are only measuring the db query count on the second try. so a different kind of test would be needed in order to test that this PR eliminates the slow startup problem that we're trying to address.

**Review by maddiedierker — APPROVED** [review]
🥇 thank you for this, will!

**Comment by wjordan** [comment]
> so a different kind of test would be needed in order to test that this PR eliminates the slow startup problem that we're trying to address.

Yeah, for covering this particular issue we need to test that the caching-initialization behavior has already cached the script on the very first request, not the second. I've added a test to cover this in 9a0326f02068a6e035b3353a265cc792eb67998b (and confirmed the test failed before this PR and passes with the fix).

**Pull request merged**